pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go

```diff
@@ -350,6 +367,829 @@ var _ = Describe("VirtualMachine Mutator", func() {
+			vm.Spec.Instancetype = instancetypeMatcher
+			vm.Spec.Preference = preferenceMatcher
+			vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+				Name: inferVolumeName,
+				VolumeSource: v1.VolumeSource{
+					PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+						PersistentVolumeClaimVolumeSource: k8sv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvc.Name,
+						},
+					},
+				},
+			})
+			vmSpec, _ := getVMSpecMetaFromResponse()
+			Expect(vmSpec.Instancetype).To(Equal(expectedInstancetypeMatcher))
+			Expect(vmSpec.Preference).To(Equal(expectedPreferenceMatcher))
+		},
+			Entry("for InstancetypeMatcher",
+				&v1.InstancetypeMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.InstancetypeMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				}, nil, nil,
+			),
+			Entry("for PreferenceMatcher",
+				nil, nil,
+				&v1.PreferenceMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.PreferenceMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				},
+			),
+		)
+
+		DescribeTable("should infer defaults from DataVolumeSource and PersistentVolumeClaim", func(instancetypeMatcher, expectedInstancetypeMatcher *v1.InstancetypeMatcher, preferenceMatcher, expectedPreferenceMatcher *v1.PreferenceMatcher) {
+			vm.Spec.Instancetype = instancetypeMatcher
+			vm.Spec.Preference = preferenceMatcher
+			vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+				Name: inferVolumeName,
+				VolumeSource: v1.VolumeSource{
+					DataVolume: &v1.DataVolumeSource{
+						Name: dvWithSourcePVC.Name,
+					},
+				},
+			})
+
+			vmSpec, _ := getVMSpecMetaFromResponse()
+			Expect(vmSpec.Instancetype).To(Equal(expectedInstancetypeMatcher))
+			Expect(vmSpec.Preference).To(Equal(expectedPreferenceMatcher))
+		},
+			Entry("for InstancetypeMatcher",
+				&v1.InstancetypeMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.InstancetypeMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				}, nil, nil,
+			),
+			Entry("for PreferenceMatcher",
+				nil, nil,
+				&v1.PreferenceMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.PreferenceMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				},
+			),
+		)
+
+		DescribeTable("should infer defaults from DataVolumeTemplate, DataVolumeSourcePVC and PersistentVolumeClaim", func(instancetypeMatcher, expectedInstancetypeMatcher *v1.InstancetypeMatcher, preferenceMatcher, expectedPreferenceMatcher *v1.PreferenceMatcher) {
+			vm.Spec.Instancetype = instancetypeMatcher
+			vm.Spec.Preference = preferenceMatcher
+			vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+				Name: inferVolumeName,
+				VolumeSource: v1.VolumeSource{
+					DataVolume: &v1.DataVolumeSource{
+						Name: "dataVolume",
```

Comment:
Make this name a const?

Reply:
I'd rather not, it's just a template name and only used within the context of this test.